flutter/apps/prompting_client_ui/linux/my_application.cc

```diff
@@ -56,18 +91,50 @@ static void my_application_activate(GApplication* application) {
   gtk_container_add(GTK_CONTAINER(window), GTK_WIDGET(view));
 
   fl_register_plugins(FL_PLUGIN_REGISTRY(view));
+
+  // Retrieve parsed arguments
+  char *snap_name = (char*)g_object_get_data(G_OBJECT(application), "snap_name");
+  guint64 app_id = GPOINTER_TO_UINT(g_object_get_data(G_OBJECT(application), "app_id"));
+
+  signal_prompting_to_gnome_shell(snap_name, app_id);
+
   gtk_widget_show(GTK_WIDGET(window));
   gtk_widget_show(GTK_WIDGET(view));
 
-  gtk_widget_grab_focus(GTK_WIDGET(view));
+  gtk_window_set_skip_taskbar_hint(window, TRUE);
+  gtk_window_set_skip_pager_hint(window, TRUE);
 }
 
 // Implements GApplication::local_command_line.
 static gboolean my_application_local_command_line(GApplication* application, gchar*** arguments, int* exit_status) {
   MyApplication* self = MY_APPLICATION(application);
+  // Parse command line arguments
+  char *snap_name = NULL;
+  guint64 app_id = 0;
+
+  GOptionEntry entries[] = {
+    { "snap", 0, 0, G_OPTION_ARG_STRING, &snap_name, "Snap name", NULL },
+    { "app-id", 0, 0, G_OPTION_ARG_INT64, &app_id, "Application ID", NULL },
```

Comment:
```suggestion
    { "app-pid", 0, 0, G_OPTION_ARG_INT64, &app_id, "Application PID", NULL },
```

It seems like there's some mild naming confusion here - what we need to pass to the shell extension is the *process ID* of the process invoking the syscall that causes the prompt. In the spec it's named `app_pid`. This is not to be confused with the ["application ID"](https://developer.gnome.org/documentation/tutorials/application-id.html) commonly used in GNOME.